calico_node/startup.py

```diff
@@ -266,7 +266,11 @@ def main():
         sys.exit(1)
 
     as_num = os.getenv("AS")
-    as_num = as_num or None
+    if os.getenv("USE_HOST_AS") == "true":
```

Comment:
@robbrockbank doesn't this make more sense?

- Use the provided AS if one exists.  Otherwise,
- Use the host AS, if it is configured.  Otherwise,
- Use the global default value.